### PR TITLE
Hash RFQ private data

### LINKED ISSUE #84: Create salted hashes of `privateData` in `Rfq`
Per https://github.com/TBD54566975/tbdex/pull/294

tbdex-js implementation: https://github.com/TBD54566975/tbdex-js/pull/215

### LINKED ISSUE #88: update rfq tests after vectors available

### PR DESCRIPTION
**Pull request by kirahsapong**
Closes #84  
Closes #88

### CODE REVIEW

**Review by diehuxx — APPROVED** [review]

File: Sources/tbDEX/Protocol/DevTools.swift [hunk 1]
```diff
@@ -80,31 +80,35 @@ enum DevTools {
...
+            )
         )
         
         if let `protocol` = `protocol` {
-            return RFQ(
+            return try RFQ(
                 to: to,
                 from: from,
-                data: rfqData,
+                rfqData: rfqData,
```
Comment (jiyoontbd) [inline Sources/tbDEX/Protocol/DevTools.swift:101]: why does this param name change from `data` to `rfqData` ?

Comment (jiyoontbd) [reply Sources/tbDEX/Protocol/DevTools.swift:101]: derp, ignore me

File: Sources/tbDEX/Protocol/Models/Messages/RFQ.swift [hunk 2]
```diff
@@ -25,10 +29,59 @@ extension RFQ {
             externalID: externalID,
             protocol: `protocol`
         )
-        self.data = data
-        self.private = nil
     }
+}
 
+private func generateSalt(_ count: Int) throws -> String? {
+    let randomBytes = [UInt8](repeating: UInt8.random(in: 0...255), count: count)
```
Comment (jiyoontbd) [inline Sources/tbDEX/Protocol/Models/Messages/RFQ.swift:36]: we're using `SecureRandom` in kotlin, could we use something equivalent to that here? 

i'm seeing [SecureRandomCopyBytes](https://developer.apple.com/documentation/security/1399291-secrandomcopybytes), maybe there's something there

File: Sources/tbDEX/Protocol/Models/Messages/RFQ.swift [hunk 3]
```diff
@@ -99,14 +152,134 @@ public struct SelectedPayoutMethod: Codable, Equatable {
     /// Type of payment method (i.e. `DEBIT_CARD`, `BITCOIN_ADDRESS`, `SQUARE_PAY`)
     public let kind: String
 
-    /// An object containing the properties defined in an Offering's `requiredPaymentDetails` json schema
+    /// A salted hash of `privateData.payin.paymentDetails`
```
Comment (jiyoontbd) [inline Sources/tbDEX/Protocol/Models/Messages/RFQ.swift:155]:
```suggestion
    /// A salted hash of `privateData.payout.paymentDetails`
```

File: Sources/tbDEX/Protocol/Models/Messages/RFQ.swift [hunk 4]
```diff
@@ -99,14 +152,134 @@ public struct SelectedPayoutMethod: Codable, Equatable {
...
+
+public struct CreateRFQPayinMethod: Codable, Equatable {
+    
+    /// Amount of payin currency you want in exchange for payout currency
+    public let amount: String
+
+    /// Type of payment method (i.e. `DEBIT_CARD`, `BITCOIN_ADDRESS`, `SQUARE_PAY`)
+    public let kind: String
+
+    /// A salted hash of `privateData.payin.paymentDetails`
```
Comment (jiyoontbd) [inline Sources/tbDEX/Protocol/Models/Messages/RFQ.swift:204]: this is not a salted hash, right?

File: Sources/tbDEX/Protocol/Models/Messages/RFQ.swift [hunk 5]
```diff
@@ -99,14 +152,134 @@ public struct SelectedPayoutMethod: Codable, Equatable {
...
         self.paymentDetails = paymentDetails
     }
 }
+
+public struct CreateRFQPayoutMethod: Codable, Equatable {
+
+    /// Type of payment method (i.e. `DEBIT_CARD`, `BITCOIN_ADDRESS`, `SQUARE_PAY`)
+    public let kind: String
+
+    /// A salted hash of `privateData.payin.paymentDetails`
```
Comment (jiyoontbd) [inline Sources/tbDEX/Protocol/Models/Messages/RFQ.swift:223]: ditto https://github.com/TBD54566975/tbdex-swift/pull/89/files#r1546503089

File: Tests/tbDEXTestVectors/tbDEXTestVectorsProtocol.swift [hunk 6]
```diff
@@ -95,7 +95,7 @@ final class tbDEXTestVectorsProtocol: XCTestCase {
         XCTAssertNoDifference(parsedQuote, vector.output)
     }
 
-    func _test_parseRfq() throws {
+    func test_parseRfq() throws {
```
Comment (jiyoontbd) [inline Tests/tbDEXTestVectors/tbDEXTestVectorsProtocol.swift:98]: aw yea

File: Tests/tbDEXTests/HttpClient/tbDEXHttpClientTests.swift [hunk 7]
```diff
@@ -125,15 +126,16 @@ final class tbDEXHttpClientTests: XCTestCase {
...
+        try balance.sign(did: altDid)
         
         Mocker.mode = .optin
         Mock(
             url: URL(string: "\(endpoint)/balances")!,
             contentType: .json,
             statusCode: 200,
             data: [
-                .get: response.data(using: .utf8)!
+                .get: try tbDEXJSONEncoder().encode(["data": [balance]])
```
Comment (jiyoontbd) [inline Tests/tbDEXTests/HttpClient/tbDEXHttpClientTests.swift:138]: just want to make sure coz this confused me tooo - are you assuming balance objects are in an array in 1 tbdex balance message?

Comment (jiyoontbd) [reply Tests/tbDEXTests/HttpClient/tbDEXHttpClientTests.swift:138]: actually, i don't think you are - this is the http response that returns
```
{
   "data": [1 tbdex message of type balance, with metadata data signature ]
}

```

so nvm!

**Review by jiyoontbd — APPROVED** [review]

**Review by nitro-neal — APPROVED** [review]

**Pull request merged**